File: dart/luzid-grpc-client/lib/src/client/meta.dart
import 'package:grpc/grpc_connection_interface.dart';
import 'package:luzid_grpc/luzid_grpc.dart';

class MetaClient {
  final MetaServiceClient _client;

  MetaClient(ClientChannelBase channel) : _client = MetaServiceClient(channel);

  Future<MetaResponse> getMeta() {
    return _client.getMeta(MetaRequest());
  }
}

File: ts/packages/luzid-grpc-client/src/client/meta.ts
import { createClient } from '@luzid/grpc-connection'
import type { Channel } from '@luzid/grpc-connection'
import {
  MetaResponse,
  MetaServiceDefinition,
  MetaServiceClient,
} from '@luzid/grpc'

export { MetaResponse, Meta } from '@luzid/grpc'

export class MetaClient {
  private readonly client: MetaServiceClient

  constructor(channel: Channel) {
    this.client = createClient(MetaServiceDefinition, channel)
  }

  /**
   * Get the meta information of the Luzid backend.
   */
  getMeta(): Promise<MetaResponse> {
    return this.client.getMeta({})
  }
}
